app/models/event.rb

```diff
@@ -3,7 +3,8 @@ class Event < ActiveRecord::Base
   before_save :calc_duration, :trim_id_code
 
   validates_presence_of :category, :title, :status, :id_code
-  validates_uniqueness_of :id_code, :title
+  validates_uniqueness_of :title
+  validates_uniqueness_of :id_code, unless: :expired?
```

Comment:
This seems awkward. I would not expect to do uniqueness on a method, especially one ending in `?`

Reply:
it's validating the uniqueness of the id_code attribute, not the method, unless it is expired - meaning the one in the database is over 6 months old. Did you not want to check for how old the one in store is?

Reply:
I'm sorry, I completely read this wrong, and I don't quite know why ! I read the `unless` as another attribute being checked, as if it were a list of three things being checked for uniqueness.
Never mind, moving on.